src/main/java/com/jcraft/jsch/Slf4jLogger.java

```diff
@@ -1,48 +1,63 @@
 package com.jcraft.jsch;
 
+import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class Slf4jLogger implements com.jcraft.jsch.Logger {
 
-  private static final org.slf4j.Logger logger = LoggerFactory.getLogger(JSch.class);
+  private static final Logger stlogger = LoggerFactory.getLogger(JSch.class);
+  private final Logger logger;
 
-  public Slf4jLogger() {}
+  public Slf4jLogger() {
+    this(stlogger);
+  }
+
+  Slf4jLogger(Logger logger) {
+    this.logger = logger;
+  }
 
   @Override
   public boolean isEnabled(int level) {
-    switch (level) {
-      case com.jcraft.jsch.Logger.DEBUG:
-        return logger.isDebugEnabled();
-      case com.jcraft.jsch.Logger.INFO:
-        return logger.isInfoEnabled();
-      case com.jcraft.jsch.Logger.WARN:
-        return logger.isWarnEnabled();
-      case com.jcraft.jsch.Logger.ERROR:
-      case com.jcraft.jsch.Logger.FATAL:
-        return logger.isErrorEnabled();
-      default:
-        return logger.isTraceEnabled();
+    switch(level) {
+        case com.jcraft.jsch.Logger.DEBUG:
+            return logger.isDebugEnabled();
+        case com.jcraft.jsch.Logger.ERROR:
+        case com.jcraft.jsch.Logger.FATAL:
+            return logger.isErrorEnabled();
+        case com.jcraft.jsch.Logger.INFO:
+            return logger.isInfoEnabled();
+        case com.jcraft.jsch.Logger.WARN:
+            return logger.isWarnEnabled();
     }
+    return logger.isTraceEnabled();
```

Comment:
I think we should put this back as a `default` case in the switch statement above, to match the switch statement in the new specialization for Throwable's below where it is in its `default` case.